File: backend/app/modules/listings/schemas/car.py
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field, ConfigDict

class CarImageBase(BaseModel):
    image_url: str
    is_primary: bool = False
    display_order: int = 0

class CarImageCreate(CarImageBase):
    pass

class CarImageResponse(CarImageBase):
    id: str
    car_id: str

    model_config = ConfigDict(from_attributes=True)

class CarBase(BaseModel):
    year: int
    mileage: int
    price: float
    vin: str = Field(..., max_length=100)
    condition: str = Field(..., max_length=50)
    fuel_type: str = Field(..., max_length=50)
    transmission: str = Field(..., max_length=50)
    exterior_color: Optional[str] = Field(None, max_length=50)
    interior_color: Optional[str] = Field(None, max_length=50)
    features: Optional[Dict[str, Any]] = None

class CarCreate(CarBase):
    model_id: str
    dealer_id: str
    images: Optional[List[CarImageCreate]] = None

class CarUpdate(BaseModel):
    price: Optional[float] = None
    mileage: Optional[int] = None
    condition: Optional[str] = Field(None, max_length=50)
    features: Optional[Dict[str, Any]] = None
    # We could allow updating other fields if needed

class CarModelMinimal(BaseModel):
    id: str
    name: str
    body_style: Optional[str] = None
    brand_id: str

    model_config = ConfigDict(from_attributes=True)

class BrandMinimal(BaseModel):
    id: str
    name: str
    slug: str
    logo_url: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

class DealerMinimal(BaseModel):
    id: str
    # In a full app this might have name, dealership_name, etc.

    model_config = ConfigDict(from_attributes=True)

class CarResponse(CarBase):
    id: str
    model_id: str
    dealer_id: str
    images: List[CarImageResponse] = []
    
    # Relationships for convenience in UI
    car_model: Optional[CarModelMinimal] = None
    brand: Optional[BrandMinimal] = None

    model_config = ConfigDict(from_attributes=True)

class CarListResponse(BaseModel):
    items: List[CarResponse]
    total: int
    skip: int
    limit: int

File: apps/web-client/src/features/listings/types/car.ts
export interface CarImage {
  id: string;
  car_id: string;
  image_url: string;
  is_primary: boolean;
  display_order: number;
}

export interface CarModelMinimal {
  id: string;
  name: string;
  body_style?: string;
  brand_id: string;
}

export interface BrandMinimal {
  id: string;
  name: string;
  slug: string;
  logo_url?: string;
}

export interface DealerMinimal {
  id: string;
}

export interface Car {
  id: string;
  model_id: string;
  dealer_id: string;
  year: number;
  mileage: number;
  price: number;
  vin: string;
  condition: string;
  fuel_type: string;
  transmission: string;
  exterior_color?: string;
  interior_color?: string;
  features?: Record<string, any>;
  images: CarImage[];
  car_model?: CarModelMinimal;
  brand?: BrandMinimal;
  dealer?: DealerMinimal;
}

export interface CarListResponse {
  items: Car[];
  total: number;
  skip: number;
  limit: number;
}
